tests/e2e/e2e_test.go

```diff
@@ -768,6 +767,38 @@ var _ = Describe("odoe2e", func() {
 
 			odoDeleteProject(projName)
 		})
+
+		It("should auto switch when deleting applications", func() {
+			// create two new projects
+
+			odoCreateProject(projName + "-auto-0")
+
+			runCmdShouldPass("odo app create app-1")
+			runCmdShouldPass("odo app create app-2")
+			runCmdShouldPass("odo app create app-3")
+
+			odoCreateProject(projName + "-auto-1")
+			runCmdShouldPass("odo app create app-4")
+			runCmdShouldPass("odo app create app-5")
+
+			// delete app in some other project which is not active
+			// the current app in the active project should not switch
+			runCmdShouldPass("odo app delete --project " + projName + "-auto-0 app-1 -f")
+			Expect(getActiveApplication()).To(Equal("app-5"))
+
+			// delete in the active project
+			// the current app should switch
+			runCmdShouldPass("odo app delete -f")
+			Expect(getActiveApplication()).To(Equal("app-4"))
+
+			// deleting the last app in the active project
+			runCmdShouldPass("odo app delete -f")
+			Expect(getActiveApplication()).To(Equal(""))
+
```

Comment:
You can add few steps to check in project ```-auto-0``` app-3 is still the active app.

For example
```
odo project -auto-0
Expect(getActiveApplication()).To(Equal("app-3"))
```

Reply:
I have created a issue https://github.com/redhat-developer/odo/issues/1363 to track of it

Reply:
Thanks for the issue.